generators/angular/support/translate-angular.ts

```diff
@@ -41,18 +41,15 @@ export type ReplacerOptions = { jhiPrefix: string; enableTranslation: boolean };
  * Replace translation key with translation values
  *
  * @param {import('../generator-base.js')} generator
+ * @param getWebappTranslation
  * @param {string} content
  * @param {string} regexSource regular expression to find keys
- * @param {object} [options]
- * @param {number} [options.keyIndex]
- * @param {number} [options.replacementIndex]
- * @param {any} [options.escape]
  * @returns {string}
  */
 function replaceTranslationKeysWithText(
-  getWebappTranslation,
-  content,
-  regexSource,
+  getWebappTranslation: (key: string, val?: Record<string, any> | undefined) => string,
```

Comment:
getWebappTranslation types should be unified.
Export from
https://github.com/jhipster/generator-jhipster/blob/9565c8060384e7003f8a2135bc9b7d4d93b0c18f/generators/base/types.d.ts#L27

Reply:
following the code, it will lead to bad behavior if set to null, and key is more meaningful than `s`.
2 choices:
- leave it as is
- ensure that method is optional elsewhere, if not change the signature

Reply:
For built in client generators it’s required but not blueprint generators.
So it’s optional every and force in built-in clients.

Reply:
True, so from Angular generator perspective, it should be mandatory